src/org/opensolaris/opengrok/index/IndexDatabase.java

```diff
@@ -969,14 +985,19 @@ private void indexDown(File dir, String parent, IndexDownArgs args)
                             }
                         }
 
-                        // If the file was not modified, skip to the next one.
-                        if (uidIter != null && uidIter.term() != null
-                                && uidIter.term().bytesEquals(buid)) {
-                            BytesRef next = uidIter.next(); // keep matching docs
-                            if (next == null) {
-                                uidIter = null;
-                            }
-                            continue;
+                        /**
+                         * If the file was not modified, probably skip to the
+                         * next one.
+                         */
+                        if (uidIter != null && uidIter.term() != null &&
+                            uidIter.term().bytesEquals(buid)) {
```

Comment:
Wrong indentation, code in condition might be mistaken for the code inside it.

It should be:
```java
if (uidIter != null && uidIter.term() != null && 
        uidIter.term().bytesEquals(buid)) {
```

Reply:
fixed